### PR TITLE
Add mozjs, V8 and WABT setup script

### PR DESCRIPTION
**Pull request by dcodeIO**
Adds a script to easily download and install mozjs, V8 and/or WABT when needed, so one doesn't have to search for them or build from source.

* mozjs reuses logic from jsvu
* V8 reuses logic from jsvu
* WABT gets the latest binaries from GitHub

Usage is

* `./third_party/setup.py mozjs` respectively
* `./third_party/setup.py v8` respectively
* `./third_party/setup.py wabt` respectively
* `./third_party/setup.py all`

Output looks like

```
$ ./third_party/setup.py all
Setting up mozjs ...
* Platform: linux-x86_64
* Latest version: 80.0b8
* Downloading to: /path/to/binaryen/third_party/mozjs
* Complete
Setting up V8 ...
* Platform: linux64
* Latest version: 8.7.3
* Downloading to: /path/to/binaryen/third_party/v8
* Complete
Setting up WABT ...
* Platform: ubuntu
* Latest release: https://github.com/WebAssembly/wabt/releases/download/1.0.19/wabt-1.0.19-ubuntu.tar.gz
* Downloading to: /path/to/binaryen/third_party/wabt/bin
* Complete
```

The respective directories are now part of `shared.py`s search paths, being preferred over global installs if set up. Tested on Linux (WSL) and Windows with Python3. Wdyt?

### CODE REVIEW

**Comment by dcodeIO** [comment]
Hmm, breakage seems unrelated.

**Comment by kripken** [comment]
The breakage is unrelated, yeah, sorry about that. It's a regression on emscripten tip of tree. It should be fixed in the current `tot` build as of right now.

**Comment by dcodeIO** [comment]
Just found that a nice side-effect of this, with all the tools right next to the build and preferred over global tools, is that one can easily copy all the things to a tmpfs for an absurdly fast fuzzing experience :)

**Comment by dcodeIO** [comment]
Btw, if this one can get some love, my fuzzing throughput would instantly multiply. (My WSL mounts live on a raid of spinning disks heh)

File: third_party/setup.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+# V8
+# see: https://github.com/GoogleChromeLabs/jsvu/tree/main/engines/v8
+
+v8_bin = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'v8')
+
+
+def v8_determine_platform():
+    is_64bits = sys.maxsize > 2**32
+    if sys.platform == 'linux' or sys.platform == 'linux2':
```
Comment (tlively) [inline third_party/setup.py:129]: Should this be `linux32` rather than `linux2`?

File: scripts/test/shared.py [hunk 2]
```diff
@@ -151,7 +151,11 @@ def is_exe(fpath):
         if is_exe(program):
             return program
     else:
-        for path in os.environ["PATH"].split(os.pathsep):
+        paths = os.environ["PATH"].split(os.pathsep)
```
Comment (tlively) [inline scripts/test/shared.py:154]: Can you add the PATH paths after the other ones to avoid having to insert paths at the beginning of the list?

**Review by tlively — APPROVED** [review]
LGTM with nits. It would probably be good to mention this script in the main README to boost its visibility, too.

File: third_party/setup.py (hunk 1 above)
Comment (dcodeIO) [reply third_party/setup.py:129]: Apparently the string used to include the major version as returned by `uname -r` prior to Python 3.3. Relevant [python docs](https://docs.python.org/3/library/sys.html#sys.platform) suggest to use `startsWith`, going to change.

**Pull request merged**